images/linux/scripts/installers/dotnetcore-sdk.sh

```diff
@@ -46,34 +46,33 @@ done
 sdks=()
 for version in ${versions[@]}; do
     release_url="https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/${version}/releases.json"
-    echo "${release_url}"
-    releases=$(curl "${release_url}")
+    download_with_retries "${release_url}" "." "${version}.json"
+    releases=$(cat "./${version}.json")
     sdks=("${sdks[@]}" $(echo "${releases}" | jq '.releases[]' | jq '.sdk.version'))
     sdks=("${sdks[@]}" $(echo "${releases}" | jq '.releases[]' | jq '.sdks[]?' | jq '.version'))
+    rm ./${version}.json
 done
 
 sortedSdks=$(echo ${sdks[@]} | tr ' ' '\n' | grep -v preview | grep -v rc | grep -v display | cut -d\" -f2 | sort -u -r)
 
+# Download additional SDKs in parallel
 for sdk in $sortedSdks; do
-    url="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$sdk/dotnet-sdk-$sdk-linux-x64.tar.gz"
-    echo "$url" >> urls
-    echo "Adding $url to list to download later"
+    (url="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$sdk/dotnet-sdk-$sdk-linux-x64.tar.gz"
+    download_with_retries $url) &
 done
+wait
```

Comment:
Just a suggestion: did you consider to parallel installation via `parallel` tool?
https://www.gnu.org/software/bash/manual/html_node/GNU-Parallel.html

Reply:
I guess it do mostly the same but include one more tool, while current option provide background processes through OS core features

Reply:
as @al-cheb mentioned, these is no child processes error handling so I will use parallel tool as it has it as build-in option